### DISCUSSION TITLE
How do I accomodate stocksplit in import?

### DISCUSSION THREAD

**Discussion by avinashkanaujiya** [ananthakumaran/paisa#33] [Q&A]
I have a zerodha tradebook, which basically have buys and sells, but no info about stocks, when I import it, it leads negative holdings in the end, due to extra selling of units courtesy of stocksplit.

**Comment by ananthakumaran** [comment]
This is something Paisa doesn't handle correctly as of today, I will get more details and report back here.

**Reply by ananthakumaran** [reply]
@avinashkanaujiya I have pushed a fix to master, would you be able to test it? You have to figure out when the split happened and need to add a transaction to represent that. Sample below, let me know if this fixes the problem

```
2002-01-01 stock split: 1 -> 2
    Assets:Stock:AAA                             -20 AAA @ 20.00 INR
    Assets:Stock:AAA                              40 AAA @ 10.00 INR
```

**Comment by sdhawade** [comment]
I believe its a bit more complicated than that, I recently did it for a few ETFs from ICICI that split, you need to find the existing buys and  "reverse" (using contra entries) the buys and insert new new buys priced at the buy price then at the appropriate ratio. This is because yahoo reports the new split price from Day 1 and not from the day of the split. Since this is complicated, I ended up using a combo of ledger cli and excel to output this file. Having native support in paisa would be great.
